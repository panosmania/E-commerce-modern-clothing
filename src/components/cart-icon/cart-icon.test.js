import { shallow } from "enzyme";
import React from "react";
import toJson from "enzyme-to-json";

import { CartIcon } from "./cart-icon.component";

describe("CartIcon component", () => {
  let wrapper;
  let mockToggleCartHidden;

  beforeEach(() => {
    mockToggleCartHidden = jest.fn();

    const mockProps = {
      itemCount: 0,
      toggleCartHidden: mockToggleCartHidden,
    };

    wrapper = shallow(<CartIcon {...mockProps} />);
  });

  it("should render CartIcon component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should call toggleCartHidden when icon is clicked", () => {
    wrapper.find(".cart-icon").simulate("click");
    expect(mockToggleCartHidden).toHaveBeenCalled();
  });

  it("should render the itemCount as the text", () => {
    const itemCount = parseInt(wrapper.find(".item-count").text());
    expect(itemCount).toBe(0);
  });
});
