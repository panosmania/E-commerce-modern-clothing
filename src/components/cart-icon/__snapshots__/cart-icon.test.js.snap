// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartIcon component should render CartIcon component 1`] = `
<div
  className="cart-icon"
  onClick={[MockFunction]}
>
  <ForwardRef(SvgShoppingBag)
    className="shopping-icon"
  />
  <span
    className="item-count"
  >
    0
  </span>
</div>
`;
